merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
I feel like this should also actually check that `ap.Leaf.Index == ap.LookupIndex`. Correct me if I'm wrong, but I think we got rid of this check entirely since we introduced `a.ProofType()`?